bootstrap/src/Main.hs

```diff
@@ -34,9 +36,15 @@ main = do
 
 main1 :: FilePath -> IO ()
 main1 planPath = do
-    meta <- I.cachedHackageMetadata
+    meta <- getMap <$> I.cachedHackageMetadata
     plan <- P.decodePlanJson planPath
     main2 meta plan
+  where
+#if MIN_VERSION_cabal_install_parsers(0,4,0)
+    getMap = snd
+#else
+    getMap = id
+#endif
```

Comment:
How about requiring `cabal-install-parsers >= 0.4`?

Reply:
Maybe, but I cannot judge the implications of taking away previously valid configurations.  I do not have any domain knowledge about the bootstrap process.
Feel free to push more commits to this PR / take it over.